File: 14.e.最长公共前缀/solution.py
#%%
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans=strs[0]
        for i in range(1,len(strs)):
            curr = strs[i]
            for j in range( min(len(ans),len(curr))  ):#
                if ans[j]!=curr[j]:
                    ans=ans[:j]
                    break
            else:
                ans=ans[:min(len(ans),len(curr)) ]
        return ans

#%%
Solution().longestCommonPrefix(strs = ["flower","flow","flight"])

# %%

File: 14.e.最长公共前缀/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int t=strs[0].length();
        string ans=strs[0];
        for (int i = 1; i < strs.size(); i++)
        {
            int j = 0;
            int m=strs[i].size();
            m = min( t, m);
            for (; j < m; j++)
            {
                if (ans[j]!=strs[i][j])
                {
                    break;
                }
            }
            t=j;
        }
        return ans.substr(0,t);
    }
};

